### ISSUE TITLE
Support variadic generics

### ISSUE THREAD

**Issue by georgiy-belyanin** [CppCXY/emmylua-analyzer-rust#75]
It'd nice to support variadic generics. Since they're used in Lua std functions it'd be handy. Take `pcall` for example. Currently only the first argument is matched.

```lua
-- False warning, expected "1" but found 10
local ok, v = pcall(string.rep, '1', 10)
```
